From 9bd08db515cdd8d82942e1dcf117daec98e7faf1 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Wed, 14 Aug 2013 15:21:30 +0800
Subject: [PATCH 4/4] ENGR00275212 enable alpha and dim layer process by
 hwcomposer

enable alpha and dim layer process by hwcomposer,
else mx6sl UI operation is very slow

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 include/hardware/hwcomposer.h      | 3 +++
 include/hardware/hwcomposer_defs.h | 6 +++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/include/hardware/hwcomposer.h b/include/hardware/hwcomposer.h
index 86479d3..ad6b5c9 100644
--- a/include/hardware/hwcomposer.h
+++ b/include/hardware/hwcomposer.h
@@ -254,6 +254,9 @@ typedef struct hwc_layer_1 {
              *   pixel.a = planeAlpha;
              *
              */
+            /* for HWC_BLENDING_NONE, alpha is not used.
+             * for HWC_BLENDING_PREMULT and HWC_BLENDING_COVERAGE, it is plane alpha value.
+             * for HWC_BLENDING_DIM, it is the alpha in source color (0,0,0,alpha). */
             uint8_t planeAlpha;
 
             /* reserved for future use */
diff --git a/include/hardware/hwcomposer_defs.h b/include/hardware/hwcomposer_defs.h
index 204faef..bf34320 100644
--- a/include/hardware/hwcomposer_defs.h
+++ b/include/hardware/hwcomposer_defs.h
@@ -108,7 +108,11 @@ enum {
     HWC_BLENDING_PREMULT  = 0x0105,
 
     /* SRC_ALPHA / ONE_MINUS_SRC_ALPHA */
-    HWC_BLENDING_COVERAGE = 0x0405
+    HWC_BLENDING_COVERAGE = 0x0405,
+
+    /* DIM. source is forced to a solid color (0,0,0,alpha), and apply
+     * (ONE / ONE_MINUS_SRC_ALPHA) blending during composition. */
+    HWC_BLENDING_DIM      = 0x0805,
 };
 
 /*
-- 
1.8.0

